packages/datetime/src/dateInput.tsx

```diff
@@ -100,10 +101,16 @@ export interface IDateInputProps extends IDatePickerBaseProps, IProps {
     /**
      * The position the date popover should appear in relative to the input box.
      * @default Position.BOTTOM
+     * @deprecated since v1.15.0, use `popoverProps.position`
      */
     popoverPosition?: Position;
 
     /**
+     * Props to pass to the `Popover`. Note that `content` cannot be changed.
+     */
+    popoverProps?: Partial<IPopoverProps>;
```

Comment:
this accepts almost any value because it's equal to `{}`. you should at least combine it with `& object` to limit it to non-primitive types.